src/reanimated2/isSharedValue.ts

```diff
@@ -1,7 +1,7 @@
 'use strict';
 import type { SharedValue } from './commonTypes';
 
-export function isSharedValue<T>(value: any): value is SharedValue<T> {
+export function isSharedValue<T>(value: unknown): value is SharedValue<T> {
   'worklet';
-  return value?._isReanimatedSharedValue === true;
+  return (value as Record<string, unknown>)?._isReanimatedSharedValue === true;
```

Comment:
`isSharedValue` is a public function and user can pass `null` or `undefined` to it so we probably shouldn't cast it to `Record` here

Reply:
Yeah, and `?.` operator will just return `undefined` in those cases

Reply:
yep but can we cast `value` which can be `null` to a `Record<string, unknown>`?

Reply:
Well, in typescript you define the types, so yes, as long as you know why you are doing it. In this case, we know that `?.` operator will handle those cases, so we can assume that if `value` isn't `Record<string, unknown>` it will evaluate to `undefined`.

Reply:
This is the best we can do here, since actually TS would like us to use `in` operator here, but we can't because of Host Objects.